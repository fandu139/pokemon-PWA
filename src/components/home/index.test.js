import React from "react";
import Render from 'react-test-renderer';
import PokemonList from './index';

// {
//   "id": "UG9rZW1vbjowMDE=",
//   "name": "Bulbasaur",
//   "number": "001",
//   "classification": "Seed PokÃ©mon",
//   "types": [
//     "Grass",
//     "Poison"
//   ],
//   "resistant": [
//     "Water",
//     "Electric",
//     "Grass",
//     "Fighting",
//     "Fairy"
//   ],
//   "attacks": {
//     "fast": [
//       {
//         "name": "Tackle",
//         "type": "Normal",
//         "damage": 12
//       },
//       {
//         "name": "Vine Whip",
//         "type": "Grass",
//         "damage": 7
//       }
//     ],
//     "special": [
//       {
//         "name": "Power Whip",
//         "type": "Grass",
//         "damage": 70
//       },
//       {
//         "name": "Seed Bomb",
//         "type": "Grass",
//         "damage": 40
//       },
//       {
//         "name": "Sludge Bomb",
//         "type": "Poison",
//         "damage": 55
//       }
//     ]
//   },
//   "weaknesses": [
//     "Fire",
//     "Ice",
//     "Flying",
//     "Psychic"
//   ],
//   "fleeRate": 0.1,
//   "maxCP": 951,
//   "evolutions": [
//     {
//       "id": "UG9rZW1vbjowMDI=",
//       "number": "002"
//     },
//     {
//       "id": "UG9rZW1vbjowMDM=",
//       "number": "003"
//     }
//   ],
//   "evolutionRequirements": {
//     "amount": 25,
//     "name": "Bulbasaur candies"
//   },
//   "maxHP": 1071,
//   "image": "https://img.pokemondb.net/artwork/bulbasaur.jpg"
// }

describe('Home Pokemon', () => {
  describe('SnaphShoot Home', () => {
    test('Home Loading', () => {
      const component = Render.create(<PokemonList />).toJSON();
      expect(component).toMatchSnapshot();
    });
  });
});
