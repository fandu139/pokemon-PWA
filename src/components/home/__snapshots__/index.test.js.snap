// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Pokemon SnaphShoot Home Home 1`] = `
Array [
  <main
    className="sc-iwsKbI iVbEbB"
  >
    <div
      className="sc-bdVaJa eTWqhe"
    >
      <div
        className="App"
      >
        <header
          className="App-header"
        >
          <div
            className="sc-bdVaJa haSTJE sc-VigVT gXCxGH"
          >
            <div
              className="lazyload-placeholder"
              style={
                Object {
                  "height": undefined,
                }
              }
            />
          </div>
          <p
            className="App-text"
          >
            Loading. Please Wait ......
          </p>
        </header>
      </div>
    </div>
  </main>,
  <footer
    className="sc-gZMcBi hAFNZL"
    data-testid="footer-component"
  >
    <div
      className="sc-bdVaJa eMKTXw sc-gqjmRU kONQNA"
    >
      <div
        className="App"
      >
        <header
          className="App-button"
        >
          <div
            className="btn-toolbar"
            role="toolbar"
          >
            <button
              className="btn btn-success btn-sm"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Filter By Type Pokemon
            </button>
          </div>
        </header>
      </div>
    </div>
  </footer>,
]
`;

exports[`Home Pokemon SnaphShoot Home Home Loading 1`] = `
Array [
  <main
    className="sc-iwsKbI iVbEbB"
  >
    <div
      className="sc-bdVaJa eTWqhe"
    >
      <div
        className="App"
      >
        <header
          className="App-header"
        >
          <div
            className="sc-bdVaJa haSTJE sc-VigVT gXCxGH"
          >
            <div
              className="lazyload-placeholder"
              style={
                Object {
                  "height": undefined,
                }
              }
            />
          </div>
          <p
            className="App-text"
          >
            Loading. Please Wait ......
          </p>
        </header>
      </div>
    </div>
  </main>,
  <footer
    className="sc-gZMcBi hAFNZL"
    data-testid="footer-component"
  >
    <div
      className="sc-bdVaJa eMKTXw sc-gqjmRU kONQNA"
    >
      <div
        className="App"
      >
        <header
          className="App-button"
        >
          <div
            className="btn-toolbar"
            role="toolbar"
          >
            <button
              className="btn btn-success btn-sm"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Filter By Type Pokemon
            </button>
          </div>
        </header>
      </div>
    </div>
  </footer>,
]
`;
